File: Binarysearch.java
public class Binarysearch
{
    public static void main(String[] args) {
        int nums [] = new int[5];
        nums[0]=1;
        nums[1]=3;
        nums[2]=5;
        nums[3]=6;
        nums[4]=9;
        
        int l=0;
        int n=9;
        int u= nums.length;
        int pos=0;
        while (l<=u) {
            int mid = (int) (l+u)/2 ;
            if(nums[mid]==n){
                pos = mid;
                System.out.println(pos);
                break;
            }
            else{
                if(n<mid){
                    u=mid-1;
                }
                else{
                    l=mid+1;
                }
            }
            
        }

    }
}

File: Binary search.py
list =[2,3,5,9,11,12]
n=int(input("Enter the Number: "))
pos=0
def binary(list,n):
    l=0
    u=len(list)-1
    while l<=u :
        mid = (l+u)//2
        if(list[mid]==n):
            globals()['pos']=mid
            print(n,"Found at:",pos)
            break
        else:
            if(list[mid]<n):
                l=mid +1 
            else:
                u=mid-1
    else:
        print("Not found",n)
        binary(list,int(input("enter another number:")))
            
binary(list,n)
            
            